src/basics.md

```diff
@@ -1242,6 +1243,50 @@ fn main() {
 }
 ```
 
+[ex-datetime-arithmetic]: #ex-datetime-arithmetic
+<a name="ex-datetime-arithmetic"></a>
+## Add or subtract dates and times using checked arithmetic
+[![chrono-badge]][chrono] [![cat-date-and-time-badge]][cat-date-and-time]
+
+Calculates and displays the date and time two weeks from now using
+[`DateTime::checked_add_signed`] and the date of yesterday using
+[`DateTime::checked_sub_signed`]. The methods return None if the date and time
+cannot be calculated.
+
+Escape sequences that are available for the
+[`DateTime::format`] can be found at [`chrono::format::strftime`].
+
+```rust
+extern crate chrono;
+use chrono::{Duration, Utc};
+
+fn main() {
+    let now = Utc::now();
+    println!("The date and time are {}", now);
+
+    if let Some(two_weeks_from_now) = now.checked_add_signed(Duration::weeks(2)) {
+        println!("Two weeks from now is {}", two_weeks_from_now);
+    } else {
+        println!("Two weeks from now overflows!");
```

Comment:
Usually errors are handled via `error_chain` crate. Not sure here it would make the code shorter. However, one way to do it is to convert the `Option` returned by `checked_add_signed` and convert it into an error via `ok_or` and use `?` to bubble up the error. Also the code has to be moved into a function `fn run() -> Result<()>`. Check other recipes for this kind of example.

Another way to improve errors is use `eprintln!` instead of `println!`.

Reply:
I did wonder a bit about chrono's decision to use `Option` here instead of Result. I think `Result` would be more correct here. But as it is, I felt that it's not appropriate to use `error_chain` with `Option` and went for the pattern matching approach.

I would love to know if my reasoning has got something to improve here. I will definitely do something about it, at least add the `eprintln!`, but do tell if you still have something to add to this.

Reply:
> I did wonder a bit about chrono's decision to use Option here instead of Result.

I agree with you. And I think it's worth to discuss it with the `chrono` team. there is [a similar talk](https://github.com/chronotope/chrono/issues/110).

> do tell if you still have something to add to this.

I though of a third solution, you keep the `if let Some` and remove the else statement. However, it's an ugly because it hides errors.

In order to be homogeneous with the rest of the cookbook and be slightly more concise, I'll go with the `error_chain` solution.

@budziq any thoughts?

Reply:
> I did wonder a bit about chrono's decision to use Option

It follows from the [stdlibs decision about checked integer arithmetic](https://doc.rust-lang.org/std/primitive.u32.html#method.checked_add). I agree that both of these decisions are debatable. 

If the `Option` `None` variant here is an `Err` is somewhat context dependent (personally I'd just go with `match` here instead of `if let Some()` or `error_chain` in the default case) but if we build some additional narrative (ie. put the calculations into a well named function indicating that it could fail) I think that we could justify `error_chain` usage. What are you thoughts?

I'd also suggest adding a series of calculations with [`and_then`](https://doc.rust-lang.org/std/option/enum.Option.html#method.and_then) combinator to spice up the example.